### ISSUE TITLE
Unable to pass true (boolean) argument as a string

### ISSUE THREAD

**Issue by gutenye** [nushell/nushell#13192]
### Describe the bug

```sh
# script.nu
def main [a: string] {
  pring $a
}
```

### How to reproduce

```sh
nu script.nu true

Error: nu::parser::parse_mismatch

  × Parse mismatch during operation.
   ╭─[<commandline>:1:6]
 1 │ main true
   ·      ──┬─
   ·        ╰── expected non-boolean value
   ╰────
```

```sh
nu script.nu (true | into string)

Error: nu::parser::parse_mismatch

  × Parse mismatch during operation.
   ╭─[<commandline>:1:6]
 1 │ main true
   ·      ──┬─
   ·        ╰── expected non-boolean value
   ╰────
```

### Expected behavior

`nu script.nu true` -> works fine

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.94.2                                          |
| major              | 0                                               |
| minor              | 94                                              |
| patch              | 2                                               |
| branch             |                                                 |
| commit_hash        |                                                 |
| build_os           | macos-aarch64                                   |
| build_target       | aarch64-apple-darwin                            |
| rust_version       | rustc 1.78.0 (9b00956e5 2024-04-29) (Homebrew)  |
| cargo_version      | cargo 1.78.0                                    |
| build_time         | 2024-06-03 07:38:55 +00:00                      |
| build_rust_channel | release                                         |
| allocator          | mimalloc                                        |
| features           | default, sqlite, system-clipboard, trash, which |
| installed_plugins  |                                                 |

**Comment by ayax79** [comment]
Good catch. I would expect that since numerical values can be coerced to a string, booleans should too.

**Comment by ayax79** [comment]
I expect this to work, but it still doesn't:

```nushell
 nu script.nu  (true | into string)
 ```

**Comment by NotTheDr01ds** [comment]
Even `nu script.nu "true"` (or `"false"`) fails.  According to the error message, it sounds like a parser issue.

**Comment by devyn** [comment]
It also doesn't seem to apply only to rest arguments, either. Weird. I swear we had code to avoid parsing args that are string-typed, so that we don't end up doing things like this. For example, `nu script.nu '"foo"'` works fine and doesn't unquote the string, so I wonder what's happening here...

**Comment by Rexcrazy804** [comment]
I am having an issue where nix shell is trying to set up the env vars for zoxide as "true" which after update 0.102.0 fails like so:

![Image](https://github.com/user-attachments/assets/979cc618-433c-44cf-9e5f-fdee1545123a)

Unsure if this is related or if I should open a new issue

**Comment by gutenye** [comment]
@Rexcrazy804 Update the line to `if (not ($env | default false __zoxide_hooked | get __zoxide_hooked | into bool)) {` in the `zoxide.nu` file

**Comment by xav-ie** [comment]
@Rexcrazy804 , @gutenye . Thank you! I was able to find out from your comment that the issue was with zoxide not knowing the latest and greatest about `nu`. 
This nix shell issue can also be solved if you are on zoxide version `0.9.7`. Then the `zoxide init nushell` script generated does not contain the issue.
